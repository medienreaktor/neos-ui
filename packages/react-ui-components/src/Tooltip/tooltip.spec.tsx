import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

import ToolTip, {TooltipProps, defaultProps} from './tooltip';

describe('<ToolTip/>', () => {
    const props: TooltipPropsÂ = {
        ...defaultProps,
        children: 'Foo children',
        theme: {
            tooltip: 'tooltipClassName',
            'tooltip--asError': 'asErrorClassName',
            'tooltip--inline': 'inlineClassName',
            'tooltip--arrow': 'arrowClassName',
            'tooltip--inner': 'innerClassName',
        },
    };

    it('should render correctly.', () => {
        const wrapper = shallow(<ToolTip {...props}/>);

        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should allow the propagation of "className" with the "className" prop.', () => {
        const wrapper = shallow(<ToolTip {...props} className="fooClassName"/>);

        expect(wrapper.prop('className')).toContain('fooClassName');
    });
});
