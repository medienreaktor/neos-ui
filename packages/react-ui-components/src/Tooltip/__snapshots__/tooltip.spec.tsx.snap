// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToolTip/> should render correctly. 1`] = `
<div
  className="tooltipClassName"
>
  <div
    className="arrowClassName"
  />
  <div
    className="innerClassName"
  >
    Foo children
  </div>
</div>
`;
